### DISCUSSION TITLE
How to make NER more consistent? It catches and misses the same entities within the same text.

### DISCUSSION THREAD

**Discussion by MrEricL** [explosion/spaCy#3952] [Help: Best practices]
I trained a NER model and I found the recall has some minor issues. 

- Companies are usually only recognized when it has a "Inc" or "Co". "Apple Inc." works fine, but "Apple" is fairly inconsistent
- There are struggles with last names. "Bob Smith" will do fine, but "Smith" is more of a hit or miss
- Certain entities like "Company X" will be tagged and not tagged

My guess is that my data is more user comments, so it's not as uniform and structured as well as a news article. With a bit more randomness, what's the best way to approach it? I've already taken into consideration`EntityRuler` but it won't help cover cases I don't expect.

**Comment by svlandeg** [comment]
In general I would advice to check the consistency of your training dataset,  e.g. if your annotations of companies / names are inconsistent in the training dataset, you will find the same in the predictions. Might be you just need more data ...

**Comment by honnibal** [comment]
In addition to @svlandeg 's advice, you might also want to check that your training is converging well. It might be that you need to fiddle with the parameters like dropout, batch size and number of epochs. It could be that your model is under-fitting, i.e. it hasn't settled on a stable theory of the data yet, and needs to keep learning.
